net/dns/resolver/tsdns.go

```diff
@@ -596,10 +611,29 @@ func (r *Resolver) resolveLocal(domain dnsname.FQDN, typ dns.Type) (netip.Addr,
 
 	r.mu.Lock()
 	hosts := r.hostToIP
+	wcHosts := r.wcHostList
 	localDomains := r.localDomains
 	r.mu.Unlock()
 
-	addrs, found := hosts[domain]
+	var found = false
+	var addrs []netip.Addr
+
+	addrs, found = hosts[domain]
+	if !found {
+		// Find matching subdomain entries
+		for _, key := range wcHosts {
+			// Entries in wcHosts already have a trailing '.', and Contains adds another
```

Comment:
you can skip `ToFQDN` as `key` is already an fqdn

Reply:
I have to convert it to a string to remove the '.' then back to an FQDN

Reply:
Are you sure, doesn't ToFQDN also perform this removal?

Reply:
Sorry I think I'm not understanding the change here. The original problem was running `Contains` off of `wilcard` since it adds a trailing '.'. So I i did remove the `[1:]` since ToFQDN does remove the '.'

Reply:
you should be able to ensure that wildcards contains FQDN values pre-sanitized in whatever way they need to be consumed here, so that no parsing or reconstruction needs to happen when answering a query.

I'm a little confused by the discussion of trailing periods and the slice prefix trim, particularly because the type has interest in both - I'm not sure which you're actually discussing.

Either way, do the necessary construction when wildcards is populated so here all you need to do during a query is check for a match.